dev_docs/key_concepts/encrypted_saved_objects.mdx

```diff
@@ -0,0 +1,194 @@
+should be included.
+
+## Caveats
+
+### Partial Updates
+
+Partial updates on ESOs are only possible if the changes are limited to unencrypted and non-AAD attributes. Any changes to an ESO's encrypted values or AAD-
+included values requires re-encryption, which means the entire object must be provided when updating to avoid corrupting the object. If an ESO is corrupted by
+a partial update, it will be effectively undecryptable. Currently, there is nothing preventing or limiting partial updates of ESOs (see open GitHub issue
+[50256](https://github.com/elastic/kibana/issues/50256)), so this requires consistent diligence from developers utilizing ESOs.
+
+### Migrations, Backward compatibility, and Serverless
+
+With time you may need to change your ESO types. Model Versions allow developers to make versioned changes to Saved Object types, but an ESO type requires special
+handling if there are changes to its encrypted attributes or attributes that are included in AAD - in both cases an object must be re-encrypted. For this case the
+ESO Service exposes a Model Version wrapper function API `createModelVersion`. Similar in utility to its predecessor (`createMigration` - used with non-Model Version
+Saved Object legacy migrations), `createModelVersion` provides a way to wrap a Model Version definition such that decryption and encryption occur automatically
+during migration of any applicable ESOs.
+
+In addition to a Model Version definition, `createModelVersion` also requires both an "input type" and "output type" `EncryptedSavedObjectTypeRegistration`
+input parameters.
+
+```ts
+export interface CreateEsoModelVersionFnOpts {
+  modelVersion: SavedObjectsModelVersion;
+  shouldTransformIfDecryptionFails?: boolean;
+  inputType: EncryptedSavedObjectTypeRegistration;
+  outputType: EncryptedSavedObjectTypeRegistration;
+}
+```
+
+The `inputType` parameter provides the necessary ESO registration definition for decrypting an ESO of the preceding Model Version prior to performing any transforms
+defined by the Model Version. The `outputType` parameter provides the necessary ESO registration definition for encrypting an ESO once the transforms defined in the
+Model Version have been completed. All of the Model Version transform functions ('unsafe_transform', 'data_backfill', 'data_removal'), are merged into a single
+transform function for efficiency. This way, each ESO only needs to be decrypted and re-encrypted once to incorporate all of the changes defined in a Model Version.
+The optional `shouldTransformIfDecryptionFails` parameter defines whether an ESO type should proceed with the Model Version changes even if decryption fails.
+
+Some examples of `createModelVersion` can be found in the ESO Model Version example plugin (
+[examples/eso_model_version_example/server/plugin.ts](https://github.com/elastic/kibana/blob/06fc22a0f15e692857ba689a7b0ddec91ed2dac2/examples/eso_model_version_example/server/plugin.ts))
```

Comment:
I would try adding the "simplest" example in this doc, and then link to the example plugin for the other examples

Reply:
Example added in 94b41de3d540d33a07c15807bb9ec922812ccae5

There were not many options to choose from to include a Model Version implemented ESO, so I just picked the simplest one, even if it was not "perfect" (we don't really have any perfectly ideal ESO definitions).